src/functions/game/onPlayedTileInteraction.ts

```diff
@@ -0,0 +1,139 @@
+import { DEFAULT_MAX_USERS_IN_GAME } from '../../utils/constants';
+import { getConnectionIdsFromUsers } from '../../utils/broadcastHelper';
+import { getUsersInGame } from '../../dynamodb/gameDBService';
+import { User } from '../../models/User';
+
+/**
+ * Compare played tile interaction and decide whose can make meld base on meld priority.
+ * Send message to all user in the game about who can take the played tile.
+ * @param {string} gameId Game Id
+ * @param {WebSocketClient} ws WebSocketClient
+ */
+export const compareTileInteractionAndSendUpdate = async (gameId: string, ws: WebSocketClient): Promise<void> => {
+  const users = (await getUsersInGame(gameId)) as User[];
+  const connectionIds = getConnectionIdsFromUsers(users);
+  const playedTileInteractions = (await getCurrentPlayedTile(gameId)) as PlayedTile[];
+  const interactions: PlayedTile[] = playedTileInteractions.filter((playedTileInteraction) => {
+    return !playedTileInteraction.skipInteraction;
+  });
+
+  let messageSent = false;
+  let connectionId: string;
+  let numOfConsecutive = 0;
+  interactions.forEach((i): unknown => {
+    const { connectionId: cid, playedTile: tile } = i;
+    const meld = i.possibleMeld as string;
+
+    const wsPayload: InteractionSuccessPayload = {
+      connectionId: cid,
+      meldType: meld,
+      playedTile: tile,
+    };
+
+    /**
+     * Making Triplet or Quad (Triplet and Quad takes precedence over Consecutive)
+     */
+    if (!messageSent && (meld === MeldEnum.TRIPLET || meld === MeldEnum.QUAD)) {
+      messageSent = true;
+      const successWsResponse = createInteractionSuccessResponse(wsPayload);
+      return Promise.all(connectionIds.map((user) => ws.send(successWsResponse, user)));
+    }
+
+    /**
+     * Making Consecutive
+     */
+    // Only one user (the next user to the user who played the tile) can make consecutive
+    if (meld === MeldEnum.CONSECUTIVE) {
+      numOfConsecutive += 1;
+      connectionId = cid;
+    }
+
+    // If no one make a Triplet or Quad, then a user can make consecutive if there is any
+    if (!messageSent && meld === MeldEnum.CONSECUTIVE && numOfConsecutive === 1) {
+      const msg = createInteractionSuccessResponse({
+        ...wsPayload,
+        connectionId,
+      });
+      return Promise.all(connectionIds.map((user) => ws.send(msg, user)));
+    }
+
+    return wsPayload;
+  });
+};
+
+export const handler: Handler = async (event: WebSocketAPIGatewayEvent): Promise<LambdaResponse> => {
+  Logger.createLogTitle('onPlayedTileInteraction.ts');
+
+  // Parse event
+  const { connectionId } = event.requestContext;
+  const body: LambdaEventBody = JSON.parse(event.body);
+  const { payload }: { payload: LambdaEventBodyPayloadOptions } = body;
+  const gameId = payload.gameId as string;
+  const playedTile = payload.playedTile as string;
+  const meldType = payload.meldType as string;
+  const skipInteraction = payload.skipInteraction as boolean;
```

Comment:
Is it possible for object destructuring as declaring type?

Reply:
sure, then it would look like this:
```ts
  const { gameId, playedTile, meldType, skipInteraction } = payload as {
    gameId: string;
    playedTile: string;
    meldType: string;
    skipInteraction: boolean;
  };
```
more lines of code 😂

Reply:
No worries then